animation_nodes/nodes/vector/vector_noise.py

```diff
@@ -1,27 +1,277 @@
+            elif self.blenderNoiseType == "3DNOISE":
+                return "execute_3dNoise"
+            elif self.blenderNoiseType == "VARIABLELACUNARITY":
+                return "execute_VariableLacunarity"
+            elif self.blenderNoiseType == "VORONOI":
+                return "execute_Voronoi"    
+
+
 
-    def execute(self, vectors, *settings):
+    def execute_FastNoise(self, vectors, *settings):
         noise = self.calculateNoise(vectors, *settings)
         return DoubleList.fromValues(noise)
+
+    #Fractal Noise Methods:
+
+    def execute_Fractal(self, vectors, fractalDimension, lacunarity ,octaves):
+        if not self.useVectorList: vectors = Vector3DList.fromValue(vectors)
+        noiseBasis = self.noiseBasis
+        values = blFractal(noiseBasis, vectors, fractalDimension, lacunarity ,octaves)
+        if not self.useVectorList:
+            return values[0]
+        else:
+            return values    
+
+    def execute_MultiFractal(self, vectors, fractalDimension, lacunarity ,octaves):
+        if not self.useVectorList: vectors = Vector3DList.fromValue(vectors)
+        noiseBasis = self.noiseBasis
+        values = blMultiFractal(noiseBasis, vectors, fractalDimension, lacunarity ,octaves)
+        if not self.useVectorList:
+            return values[0]
+        else:
+            return values
+
+    def execute_HeteroTerrain(self, vectors, fractalDimension, lacunarity ,octaves, offset):
+        if not self.useVectorList: vectors = Vector3DList.fromValue(vectors)
+        noiseBasis = self.noiseBasis
+        values = blHeteroTerrain(noiseBasis, vectors, fractalDimension, lacunarity ,octaves, offset)
+        if not self.useVectorList:
+            return values[0]
+        else:
+            return values
+
+    def execute_RigidMultiFractal(self, vectors, fractalDimension, lacunarity ,octaves, offset, gain):
+        if not self.useVectorList: vectors = Vector3DList.fromValue(vectors)
+        noiseBasis = self.noiseBasis
+        values = blRigidMultiFractal(noiseBasis, vectors, fractalDimension, lacunarity ,octaves, offset, gain)
+        if not self.useVectorList:
+            return values[0]
+        else:
+            return values
+
+    def execute_HybridMultiFractal(self, vectors, fractalDimension, lacunarity ,octaves, offset, gain):
+        if not self.useVectorList: vectors = Vector3DList.fromValue(vectors)           
+        noiseBasis = self.noiseBasis
+        values = blHybridMultiFractal(noiseBasis, vectors, fractalDimension, lacunarity ,octaves, offset, gain)
+        if not self.useVectorList:
+            return values[0]
+        else:
+            return values 
+
+    #3D Noise Methods:
+
+    def execute_3dNoise(self, vectors, seed, octaves, hard, amplitude, frequency):
+        if not self.useVectorList: vectors = Vector3DList.fromValue(vectors) 
+        noiseBasis = self.noiseBasis
+        values = blTurbulence(noiseBasis, vectors, seed, octaves, hard, amplitude, frequency)
```

Comment:
Make the non-zero value for seed `max(seed, 1)` here, and remove that from the `c_utils.pyx` file.